#!/usr/bin/env pyston3

import argparse
import logging
import os
import os.path
import sys
import traceback

from solana.publickey import PublicKey

sys.path.insert(0, os.path.abspath(
    os.path.join(os.path.dirname(__file__), '..')))
import mango  # nopep8

# We explicitly want argument parsing to be outside the main try-except block because some arguments
# (like --help) will cause an exit, which our except: block traps.
parser = argparse.ArgumentParser(description="Shows the on-chain data of a Mango Markets Margin Account.")
mango.Context.add_command_line_parameters(parser)
parser.add_argument("--address", type=PublicKey, required=True,
                    help="Solana address of the Mango Markets Margin Account")
args = parser.parse_args()

logging.getLogger().setLevel(args.log_level)
logging.warning(mango.WARNING_DISCLAIMER_TEXT)

try:
    context = mango.Context.from_command_line_parameters(args)

    group = mango.Group.load(context)
    margin_account = mango.MarginAccount.load(context, args.address, group)

    print(margin_account)
except Exception as exception:
    logging.critical(f"group-balances stopped because of exception: {exception} - {traceback.format_exc()}")
except:
    logging.critical(f"group-balances stopped because of uncatchable error: {traceback.format_exc()}")
